edu cluster overlay

jtriley opened the pull request:
This is a copy of #726 with the infra rebuild commit removed and pushed to nerc-ocp-config repo for collaboration. Copying the description here for reference:


From @tssala23:
```
Created with my templating playbook and then modified to reflect prod
There are some comments for things that may have to be changed or that I updated
I made a couple assumptions in this:

    the address will be edu.nerc.mghpcc.org
    the vault secret path will be nerc-ocp-edu

Keycloack and github id need to be added
Machine configs need to be added
nncp need to be added
The oauth dir seems to be redundant
```

tssala23 committed:
Add overlay for edu cluster

Signed-off-by: tssala23 <tsalawu@redhat.com>

jtriley committed:
edu: add udev + disable-net-ifnames machineconfigs

jtriley committed:
edu: add clusterversion manifest

jtriley committed:
edu: add nmstate configs for nese and infra

jtriley committed:
edu: add nfd-operator overlay

jtriley committed:
edu: add fake-metrics-server overlay

jtriley committed:
edu: set github idp clientID

jtriley committed:
edu: remove xdmod-reader bundle

jtriley committed:
edu: add rhoai datasciencecluster

jtriley committed:
edu: downgrade rhoai to stable-2.19

jtriley committed:
edu: update odf rbd storageclass pool

jtriley committed:
edu: bump aap operator to stable-2.5-cluster-scoped

jtriley committed:
edu: remove unavailable elasticsearch-operator

This operator is no longer installable on 4.19. Most likely was
deprecated. Either we live without it or find another operator/solution.

tssala23 reviewed (approved):
Gave it a scan, no obvious issues to me. There is a comment I left in the kustomization file line 137 that could be removed but wont cause any issues if not obviously

naved001 commented on cluster-scope/overlays/edu/kustomization.yaml:
```diff
@@ -0,0 +1,172 @@
+- ../../bundles/koku-metrics-operator
+- ../../bundles/autopilot
+- ../../bundles/mongodb-operator
+- ../../bundles/nagios-monitoring
```
@jtriley can we add the [xdmod-reader bundle](https://github.com/OCP-on-NERC/nerc-ocp-config/tree/main/cluster-scope/bundles/xdmod-reader)? it'll create a serviceaccount and a token that I can use for gathering metrics from the cluster. Thanks!

tssala23 replied:
Funny because one of the commits removes that bundle https://github.com/OCP-on-NERC/nerc-ocp-config/pull/727/commits/6719f5ba7db881e4816f7e7227d599c99c0a12eb

naved001 replied:
Good catch! I guess the name is kind of misleading as it has nothing to do with xdmod (we wanted to use it, but then never used it). The bundle is just some rbacs, a service account and a token to collect metrics from prometheus/thanos

jtriley replied:
@naved001 Let's rename it instead - what would be a representative name?

naved001 replied:
`moc-billing` or `moc-invoicing` or `moc-metrics-collector`. Either of those is fine.

jtriley replied:
OK let's go with `moc-metrics-collector`. I'm going to merge this and do that in a separate PR after I finish the coldfront portion of today's maintenance.

jtriley committed:
edu: disable object-storage manifests for now

Need to check with the team whether this is needed at all

jtriley committed:
edu: rename overlay to nerc-ocp-edu

naved001 commented on cluster-scope/overlays/nerc-ocp-edu/clusterrolebindings/autopilot-service-acc-node-reader.yaml:
```diff
@@ -0,0 +1,12 @@
+subjects:
+  - kind: ServiceAccount
+    name: autopilot-service-acc
+    namespace: autopilot-dashboard-f3dc9e
```
Not sure if it matters, but I don't see where this namespace is created. This seems to be a coldfront generated name.

The namespace created by the auotpilot [bundle](https://github.com/OCP-on-NERC/nerc-ocp-config/blob/main/cluster-scope/base/core/namespaces/autopilot/namespace.yaml) is `autopilot`.

jtriley replied:
Most likely was just copy/pasted from prod overlay would be my guess. Not sure the back story on that one.

jtriley replied:
I've removed them from the cluster and this overlay for now. We can add them back later if they're still needed on this new edu/academic cluster.

naved001 reviewed (approved):
Just one comment, but looks fine to me.

jtriley committed:
edu: add acct-mgt overlay

We only need the service account token for coldfront after today's
maintenance (20250812) given that it no longer relies on the acct-mgt
deployment for configuring projects/users on OpenShift.

jtriley committed:
edu: remove clusterrrolebindings for now

These were added to ner-ocp-prod cluster for individual users/projects
and aren't currently relevant for the nerc-ocp-edu cluster. We can add
these back in the future if they're still required on the new
edu/academic cluster.
